best-practices-in-kubernetes-security.md

```diff
@@ -0,0 +1,180 @@
+
+### **Use Process Whitelisting**
+
+Process whitelisting is an efficient Kubernetes security technique that provides zero-trust guarantee as far as process security is concerned. You should create a list of processes that are allowed to run on each pod. Next, you should use this list as your application's whitelist.
+
+Suppose an attacker succeeds to get access to your cluster and begins running a malicious process. In that case, you will easily identify the user since it is a new non-whitelisted process. If it isn't on the whitelist and isn't part of the pod's normal behavior either, it should be reported immediately.
+
+### **Deploy Additional Controls**
+
+If you're using Kubernetes in a production environment, it is imperative that you deploy additional security controls. The reason is that you need complete visibility into the Kubernetes environment and the activities going on there. This would help you to avoid potential downtimes when you've to deploy mission-critical applications.
+
+This would help you to avoid potential downtimes when deploying mission-critical applications. These controls exist as policies to enable you to restrict access to resources in the cluster.
+
+### **Monitor Network Traffic**
+
+Ideally, you should keep an eye on network activity and compare it to authorized traffic and traffic permitted by the Kubernetes network policy. The goal is to eliminate connections that are no longer required in order to minimize the attack surface. Additionally, you may strengthen the permitted network policy by removing any connections that aren't absolutely required.
+
+### **Adopt a Multi-Layered Security Approach**
+
+Cloud-native applications have diverse components and elements, making it imperative to look beyond containerized workloads to safeguard your Kubernetes deployments. Hence to secure your Kubernetes environment, you should adopt a multi-layered security strategy. The Kubernetes documentation recommends a multi-layered approach called 4C (Cloud, Container, Cluster and Cloud) for securing your Kubernetes environments.
+
+-   Cloud and data center infrastructure and other resources required to create Kubernetes environments
+
+-   Container clusters that comprise the workloads that run on a particular cluster
+
+-   Individual Containers, i.e., the software images used to instantiate workloads.
+
+-   Source Code and container images
+
+It would help if you took advantage of recommended tools to secure and protect each of these layers. However, you should know that you don't have one tool that can help protect all these layers.
+
+### **Enable Kubernetes Role-Based Access Control**
```

Comment:
There's another item of the list with the same name at the beginning of the most. Are they different?

Reply:
Oops! Sorry. This section is redundant, I've removed the text from one and checked-in the draft again.